test/src/vec_test.cpp

```diff
@@ -218,6 +218,13 @@ namespace vm {
         ASSERT_EQ(+1, compare(vec3f(1, 1, 2), vec3f::one));
         ASSERT_EQ(+1, compare(vec3f(2, 0, 0), vec3f::one));
         ASSERT_EQ(+1, compare(vec3f(1, 2, 0), vec3f::one));
+
+        ASSERT_NE( 0, compare(vec3f(1, 2, 0), vec3f::NaN));
+        ASSERT_NE( 0, compare(vec3f::NaN,     vec3f(1, 2, 0)));
```

Comment:
If I understand correctly, the problem is that we have a compare function, right? The function cannot return a value that is consistent with all of the operators defined on it because there is no generic "unordered" value. But what if there was? We could tighten compare to return -1, 0, and +1 for less, equal, greater, and +2 for unordered or something. Or, even better, handle the NaN case directly in the operators before calling the compare function.

Or am I missing something?

Reply:
Ok!